executor/analyze.go

```diff
@@ -517,6 +522,33 @@ func analyzeColumnsPushdown(colExec *AnalyzeColumnsExec) []analyzeResult {
 	} else {
 		ranges = ranger.FullIntRange(false)
 	}
+	if colExec.analyzeVer == statistics.Version3 {
+		count, hists, topns, fmSketches, err := colExec.buildSamplingStats(ranges)
+		if err != nil {
+			return []analyzeResult{{Err: err, job: colExec.job}}
+		}
+		cLen := len(colExec.analyzePB.ColReq.ColumnsInfo)
+		colResult := analyzeResult{
+			TableID:  colExec.tableID,
+			Hist:     hists[:cLen],
+			TopNs:    topns[:cLen],
+			Fms:      fmSketches[:cLen],
+			job:      colExec.job,
+			StatsVer: colExec.analyzeVer,
+			Count:    count,
+		}
+		idxResult := analyzeResult{
+			TableID:  colExec.tableID,
+			Hist:     hists[cLen:],
+			TopNs:    topns[cLen:],
+			Fms:      fmSketches[cLen:],
+			job:      colExec.job,
+			StatsVer: colExec.analyzeVer,
+			Count:    count,
+			IsIndex:  1,
+		}
+		return []analyzeResult{colResult, idxResult}
```

Comment:
Return index stats in `analyzeColumnsPushdown` looks weird.

Reply:
Or we change the table to `analyzeTablePushDown`?

Reply:
Changed the result to `colGroupResult`.